File: Pascal's_triangle.py
def generate_pascal_triangle(height):
    triangle = []
    for i in range(height):
        row = [1] * (i+1)  # генерираме реда със съответната дължина и запълваме с 1
        triangle.append(row)

    for i in range(2, height):
        row = triangle[i]
        for j in range(1, i):
            row[j] = triangle[i-1][j-1] + triangle[i-1][j]

    return triangle

def print_pascal_triangle(triangle):
    max_number_width = len(str(triangle[-1][len(triangle[-1]) // 2]))  # определяме ширината на средното число на последния ред
    height = len(triangle)
    for i in range(height):
        row = triangle[i]
        spaces = " " * (height - i - 1)  # добавяме интервали преди числата
        numbers = " ".join(str(number) for number in row)  # превръщаме числата в низ
        print(spaces + numbers)

height = int(input("Въведете височина на Триъгълника на Паскал: "))
triangle = generate_pascal_triangle(height)
print_pascal_triangle(triangle)

File: Pascal's triangle.cpp
#include <iostream>
#include <vector>
#include <string>

std::vector<std::vector<int>> generate_pascal_triangle(int height) {
    std::vector<std::vector<int>> triangle;
    for (int i = 0; i < height; i++) {
        std::vector<int> row(i + 1, 1);
        triangle.push_back(row);
    }

    for (int i = 2; i < height; i++) {
        std::vector<int>& row = triangle[i];
        for (int j = 1; j < i; j++) {
            row[j] = triangle[i - 1][j - 1] + triangle[i - 1][j];
        }
    }

    return triangle;
}

void print_pascal_triangle(const std::vector<std::vector<int>>& triangle) {
    int max_number_width = std::to_string(triangle.back()[triangle.back().size() / 2]).length();
    int height = triangle.size();
    for (int i = 0; i < height; i++) {
        const std::vector<int>& row = triangle[i];
        std::string spaces(height - i - 1, ' ');
        std::string numbers;
        for (int number : row) {
            numbers += std::to_string(number) + " ";
        }
        std::cout << spaces + numbers << std::endl;
    }
}

int main() {
    int height;
    std::cout << "Въведете височина на Триъгълника на Паскал: ";
    std::cin >> height;

    std::vector<std::vector<int>> triangle = generate_pascal_triangle(height);
    print_pascal_triangle(triangle);

    return 0;
}
